src/UI/QuickmenuView.re

```diff
@@ -60,83 +56,103 @@ module Styles = {
     ];
 };
 
+
+let onSelectedChange = index =>
+  GlobalContext.current().dispatch(MenuFocus(index));
+
+let onInput = (text, cursorPosition) =>
+  GlobalContext.current().dispatch(MenuInput({ text, cursorPosition }));
+
+let onSelect = (_) =>
+  GlobalContext.current().dispatch(MenuSelect);
+
+
 let createElement =
     (
       ~children as _,
       ~font: Types.UiFont.t,
       ~theme: Theme.t,
       ~configuration: Configuration.t,
       ~autofocus: bool=true,
-      ~prefix: option(string)=?,
+      ~state: Quickmenu.t,
       ~placeholder: string="type here to search the menu",
-      ~loadingAnimation: Animation.t=Animation.create(~isActive=false, ()),
-      ~text: string,
-      ~cursorPosition: int,
-      ~items: items,
-      ~selected: option(int),
-      ~onInput: (string, int) => unit,
-      ~onSelectedChange: int => unit,
-      ~onSelect: int => unit,
+      ~onInput: (string, int) => unit = onInput,
+      ~onSelectedChange: int => unit = onSelectedChange,
+      ~onSelect: int => unit = onSelect,
       (),
     ) =>
   component(hooks => {
+    let Quickmenu.{source, selected, text, cursorPosition, prefix} = state;
+
+    // TODO: Cam this be removed in favor of NotifyKeyPressed in MenuStoreConnector?
     let handleKeyDown = (event: NodeEvents.keyEventParams) =>
       switch (event) {
       | {key: Revery.Key.KEY_DOWN, _} =>
-        GlobalContext.current().dispatch(MenuNextItem)
+        GlobalContext.current().dispatch(MenuFocusNext)
       | {key: Revery.Key.KEY_UP, _} =>
-        GlobalContext.current().dispatch(MenuPreviousItem)
+        GlobalContext.current().dispatch(MenuFocusPrevious)
       | _ => ()
       };
 
     let (items, jobProgress) =
-      switch (items) {
-        | Array(items) =>
+      switch (source) {
+        | Loading =>
+          ([||], 0.)
+
+        | Progress({ items, progress }) =>
+          (items, progress)
+
+        | Complete(items) =>
           (items, 1.)
-        | Job(job) =>
-          (Job.getCompletedWork(job).uiFiltered, Job.getProgress(job))
       };
 
-    let time = Time.getTime() |> Time.to_float_seconds;
-
-    let loadingOpacityAnimation = Animation.getValue(loadingAnimation);
     let loadingSpinner =
-      Animation.isActive(loadingAnimation)
-        ? <View style=Style.[height(40), width(Constants.menuWidth)]>
-            <Center>
-              <View
-                style=Style.[
-                  transform(
-                    Transform.[RotateY(Math.Angle.Radians(time *. 2.))],
+      if (jobProgress == 0.) {
```

Comment:
Ya, sorry, it is more complicated than it should be! There was a bug previously with `OnMount` effects: https://github.com/briskml/brisk-reconciler/pull/40 - this would cause the `OnMount` cleanup function to not be called.... which would leave animations hanging (triggering re-renders, and also leaking memory).

But now that this is fixed - we could go back to a declarative approach and animate the components and remove the animations from being handled.

There were two animations that I used for the 'quickmenu`:
- A progress bar at the bottom of the menu, while we were discovering files.
- A spinner while we were filtering files.

Reply:
Two animations still seem needlessly complicated though. I understand they say different things of course, but I don't think end-users do, much less which process they refer to. We should either try to unify them into one animation, or just go with a basic "working" animation.